File: services/go/services/v2/echo_service.go
package v2

import (
	"errors"
	"spec/v2/echo"
	"spec/v2/models"
)

type EchoService struct{}

func (service *EchoService) EchoBody(body *models.Message) (*echo.EchoBodyResponse, error) {
	return nil, errors.New("implementation has not added yet")
}

File: services/ts-koa-superstruct/services/v2/echo_service.ts
import * as service from './../../v2/echo_service'
import * as models from './../../v2/models'

export let echoService = (): service.EchoService => {
    let echoBody = async (params: service.EchoBodyParams): Promise<models.Message> => {
        throw new Error('Not Implemented')
    }

    return {echoBody}
}
